test/e2e/gcp_test.go

```diff
@@ -0,0 +1,207 @@
+		},
+		Data: map[string][]byte{
+			"gcp-credentials.json": []byte(g.gcpCredentials),
+		},
+	}
+}
+
+func (g *gcpTestHelper) platform() string {
+	return string(configv1.GCPPlatformType)
+}
+
+func (g *gcpTestHelper) ensureHostedZone() (string, []string, error) {
+	gcpRootDomain := hostedZoneDomain + "."
+
+	resp, err := g.dnsService.ManagedZones.List(g.gcpProjectId).Do()
+	if err != nil {
+		return "", nil, fmt.Errorf("failed to list managed zones: %w", err)
+	}
+	zones := resp.ManagedZones
+	// if managed zone exists then return its id and nameservers
+	for _, zone := range zones {
+		if zone.DnsName == gcpRootDomain {
+			return strconv.FormatUint(zone.Id, 10), zone.NameServers, nil
+		}
+	}
+
+	// if zone does not exist, create managed zone and return its id and nameservers
+	zone, err := g.dnsService.ManagedZones.Create(g.gcpProjectId, &dns.ManagedZone{
+		// must be 1-63 characters long, must begin with a letter,
+		// end with a letter or digit, and only contain lowercase letters, digits or dashes
+		Name:        "a" + strings.ToLower(strings.ReplaceAll(hostedZoneDomain, ".", "-")),
+		DnsName:     gcpRootDomain,
+		Description: "ExternalDNS Operator test managed zone.",
+	}).Do()
+	if err != nil {
+		return "", nil, fmt.Errorf("error creating managed zone: %w", err)
+	}
+	return strconv.FormatUint(zone.Id, 10), zone.NameServers, nil
+}
+
+func (g *gcpTestHelper) deleteHostedZone() error {
+	resp, err := g.dnsService.ResourceRecordSets.List(g.gcpProjectId, hostedZoneID).Do()
+	if err != nil {
+		return fmt.Errorf("failed to retrieve dns records for zoneID %v: %w", hostedZoneID, err)
+	}
+
+	recordChanges := &dns.Change{}
+
+	// create change set deleting all DNS records which are not of NS and SOA types in managed zone
+	for len(resp.Rrsets) != 0 {
+		for _, recordset := range resp.Rrsets {
+			if recordset.Type == "SOA" || recordset.Type == "NS" {
+				continue
+			} else {
+				recordChanges.Deletions = append(recordChanges.Deletions, recordset)
+			}
+		}
+		if resp.NextPageToken == "" {
+			break
+		} else {
+			token := resp.NextPageToken
+			resp, err = g.dnsService.ResourceRecordSets.List(g.gcpProjectId, hostedZoneID).PageToken(token).Do()
+			if err != nil {
+				return fmt.Errorf("failed to retrieve dns records for zoneID %v and pageToken %v: %w", hostedZoneID, token, err)
+			}
+		}
+	}
+
+	// atomically delete the records from the managed zone
+	if len(recordChanges.Deletions) != 0 {
+		if _, err := g.dnsService.Changes.Create(g.gcpProjectId, hostedZoneID, recordChanges).Do(); err != nil {
+			return fmt.Errorf("failed to execute the change operation for records deletion: %w", err)
+		}
+	}
+
+	// delete the managed zone itself
+	err = g.dnsService.ManagedZones.Delete(g.gcpProjectId, hostedZoneID).Do()
+	if err != nil {
+		return fmt.Errorf("failed to delete zone with id %v: %w", hostedZoneID, err)
+	}
+
+	return nil
+}
+
+func prepareGCPConfigurations(openshiftCI bool) (gcpCredentials, gcpProjectId string, err error) {
```

Comment:
Just for the sake of consistency (and ease of maintenance which comes with it), let's keep the same name for all the providers: `prepareConfigurations`.

Reply:
same name not possible since these are not member function,
we have to bring either data members or different package?
i suggest data member if name consistency